### DISCUSSION TITLE
ValueError: Failed to load delegate from libedgetpu.so.1.0

### DISCUSSION THREAD

**Discussion by tpbillund** [blakeblackshear/frigate#1053] [General]
Hi

Im trying to install frigate on my synology with:
`sudo docker run -itd --name=frigate --restart=always --privileged -p 172.25.100.201:5005:5000 -p 172.25.100.201:1935:1935 --shm-size=1024m --device /dev/dri:/dev/dri -v /dev/bus/usb:/dev/bus/usb -v /volume1/docker/frigate/media:/media/frigate -v /volume1/docker/frigate/config:/config:ro -v /etc/localtime:/etc/localtime:ro -e FONTCONFIG_PATH='/etc/fonts' -e FRIGATE_RTSP_PASSWORD='password' blakeblackshear/frigate:stable-amd64`

And when I look at my log in portainer, I get this loop code/error, can someone help?

> Starting migrations
> 
> peewee_migrate                 INFO    : Starting migrations
> 
> Migrate "001_create_events_table"
> 
> peewee_migrate                 INFO    : Migrate "001_create_events_table"
> 
> sql ('CREATE TABLE IF NOT EXISTS "event" ("id" VARCHAR(30) NOT NULL PRIMARY KEY, "label" VARCHAR(20) NOT NULL, "camera" VARCHAR(20) NOT NULL, "start_time" DATETIME NOT NULL, "end_time" DATETIME NOT NULL, "top_score" REAL NOT NULL, "false_positive" INTEGER NOT NULL, "zones" JSON NOT NULL, "thumbnail" TEXT NOT NULL)',)
> 
> peewee_migrate                 INFO    : sql ('CREATE TABLE IF NOT EXISTS "event" ("id" VARCHAR(30) NOT NULL PRIMARY KEY, "label" VARCHAR(20) NOT NULL, "camera" VARCHAR(20) NOT NULL, "start_time" DATETIME NOT NULL, "end_time" DATETIME NOT NULL, "top_score" REAL NOT NULL, "false_positive" INTEGER NOT NULL, "zones" JSON NOT NULL, "thumbnail" TEXT NOT NULL)',)
> 
> sql ('CREATE INDEX IF NOT EXISTS "event_label" ON "event" ("label")',)
> 
> peewee_migrate                 INFO    : sql ('CREATE INDEX IF NOT EXISTS "event_label" ON "event" ("label")',)
> 
> sql ('CREATE INDEX IF NOT EXISTS "event_camera" ON "event" ("camera")',)
> 
> peewee_migrate                 INFO    : sql ('CREATE INDEX IF NOT EXISTS "event_camera" ON "event" ("camera")',)
> 
> Done 001_create_events_table
> 
> peewee_migrate                 INFO    : Done 001_create_events_table
> 
> Migrate "002_add_clip_snapshot"
> 
> peewee_migrate                 INFO    : Migrate "002_add_clip_snapshot"
> 
> add_column ('event', 'has_clip', <BooleanField: Event.has_clip>)
> 
> peewee_migrate                 INFO    : add_column ('event', 'has_clip', <BooleanField: Event.has_clip>)
> 
> add_column ('event', 'has_snapshot', <BooleanField: Event.has_snapshot>)
> 
> peewee_migrate                 INFO    : add_column ('event', 'has_snapshot', <BooleanField: Event.has_snapshot>)
> 
> Done 002_add_clip_snapshot
> 
> peewee_migrate                 INFO    : Done 002_add_clip_snapshot
> 
> frigate.mqtt                   INFO    : MQTT connected
> 
> detector.coral                 INFO    : Starting detection process: 45
> 
> frigate.app                    INFO    : Camera processor started for foran: 48
> 
> frigate.app                    INFO    : Capture process started for foran: 49
> 
> frigate.edgetpu                INFO    : Attempting to load TPU as usb
> 
> frigate.edgetpu                INFO    : No EdgeTPU detected.
> 
> Process detector:coral:
> 
> Traceback (most recent call last):
> 
>   File "/usr/local/lib/python3.8/dist-packages/tflite_runtime/interpreter.py", line 152, in load_delegate
> 
>     delegate = Delegate(library, options)
> 
>   File "/usr/local/lib/python3.8/dist-packages/tflite_runtime/interpreter.py", line 111, in __init__
> 
>     raise ValueError(capture.message)
> 
> ValueError
> 
> 
> 
> During handling of the above exception, another exception occurred:
> 
> 
> 
> Traceback (most recent call last):
> 
>   File "/usr/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
> 
>     self.run()
> 
>   File "/usr/lib/python3.8/multiprocessing/process.py", line 108, in run
> 
>     self._target(*self._args, **self._kwargs)
> 
>   File "/opt/frigate/frigate/edgetpu.py", line 124, in run_detector
> 
>     object_detector = LocalObjectDetector(tf_device=tf_device, num_threads=num_threads)
> 
>   File "/opt/frigate/frigate/edgetpu.py", line 63, in __init__
> 
>     edge_tpu_delegate = load_delegate('libedgetpu.so.1.0', device_config)
> 
>   File "/usr/local/lib/python3.8/dist-packages/tflite_runtime/interpreter.py", line 154, in load_delegate
> 
>     raise ValueError('Failed to load delegate from {}\n{}'.format(
> 
> ValueError: Failed to load delegate from libedgetpu.so.1.0
> 
> 
> 
> frigate.watchdog               INFO    : Detection appears to have stopped. Exiting frigate...
> 
> frigate.app                    INFO    : Stopping...
> 
> frigate.events                 INFO    : Exiting event cleanup...
> 
> frigate.events                 INFO    : Exiting event processor...
> 
> frigate.record                 INFO    : Exiting recording maintenance...
> 
> frigate.object_processing      INFO    : Exiting object processor...
> 
> frigate.watchdog               INFO    : Exiting watchdog...
> 
> frigate.stats                  INFO    : Exiting watchdog...
> 
> peewee.sqliteq                 INFO    : writer received shutdown request, exiting.
> 
> root                           INFO    : Waiting for detection process to exit gracefully...
> 
> frigate.video                  INFO    : foran: exiting subprocess
> 
>  * Starting nginx nginx        
> [ OK ]
> 
> frigate.app                    WARNING : Camera foran has rtmp enabled, but rtmp is not assigned to an input.
> 
> Starting migrations
> 
> peewee_migrate                 INFO    : Starting migrations
> 
> There is nothing to migrate
> 
> peewee_migrate                 INFO    : There is nothing to migrate
> 
> frigate.mqtt                   INFO    : MQTT connected
> 
> detector.coral                 INFO    : Starting detection process: 41
> 
> frigate.app                    INFO    : Camera processor started for foran: 44
> 
> frigate.app                    INFO    : Capture process started for foran: 45
> 
> frigate.edgetpu                INFO    : Attempting to load TPU as usb
> 
> frigate.edgetpu                INFO    : No EdgeTPU detected.
> 
> Process detector:coral:

**Comment by blakeblackshear** [comment]
Do you have a Coral device? If not, you need to configure CPU detectors.

**Reply by tpbillund** [reply]
@blakeblackshear 
Ahh okay I missed that and thx.

The Coral device will give me less CPU load on the Synology NAS ?

About the Proxmox code, can you maybe update that section, when you have time... about how to run it inside Proxmox, can't see how to use the "script" that you wrote https://blakeblackshear.github.io/frigate/installation/#proxmox
Have a NUC and installed Proxmox, but can't see how to add the Frigate by the "script" :-)

But so far, great job !

**Comment by tpbillund** [comment]
INFO! (To other with a Synology DS1618+

Running 1x2K camera, with clips and detection (only person) - Load without detection
![frigate_1](https://user-images.githubusercontent.com/7550184/117101266-18dfd300-ad76-11eb-8a24-189cd9257bf5.PNG)

Running 2x2K camera, with clips and detection (only person) - Load without detection
![frigate_2](https://user-images.githubusercontent.com/7550184/117101281-209f7780-ad76-11eb-9b9c-873084dd6880.PNG)

Running 3x2K camera, with clips and detection (only person) - Load without detection
![frigate_3](https://user-images.githubusercontent.com/7550184/117101294-29904900-ad76-11eb-8f1d-abe9aebea787.PNG)

And then the load get a boost of 25-30% when detecting persons.

**Comment by blakeblackshear** [comment]
A Coral device is strongly recommended. They are inexpensive and very efficient. CPU detectors are really only recommended for testing.

**Comment by tpbillund** [comment]
Fine I have ordered one now :-)
THX.
